lib/model/folder_sendrecv.go

```diff
@@ -291,6 +291,8 @@ func (f *sendReceiveFolder) pullerIteration(scanChan chan<- string) int {
 }
 
 func (f *sendReceiveFolder) processNeeded(dbUpdateChan chan<- dbUpdateJob, copyChan chan<- copyBlocksState, scanChan chan<- string) (int, map[string]protocol.FileInfo, []protocol.FileInfo, error) {
+	defer f.queue.Reset()
```

Comment:
This seems to be cause problems now, as I'm explaining [here](https://github.com/syncthing/syncthing/issues/6219#issuecomment-570759947), as `/rest/db/need` call relies on the queue to be up-to-date all the time, not only why processed by `processNeeded`. Is there a reason we need "defer" here at all, and not just plain `f.queue.Reset` so that it's done before the processing and not after? This way the queue will be correct most of the time.  @imsodin @calmh

Reply:
what you describe is the intended behaviour and i don't see a problem: progress can only be shown while something is being processed. at the point queue.Reset is called there's anyway nothing in queue anymore.

Reply:
@imsodin processNeeded finishes much earlier that the pullers finished their job.

Reply:
Right, by default the last 2 file are indeed affected by this. i think a fix should be straight forward (just reset after the finisher routine stops). I'll have a look tomorrow.

Reply:
Sounds good to me. Thank you for looking into this.